bot.ml

```diff
@@ -786,15 +786,158 @@ let callback _conn req body =
+                                  ~commit_body:
+                                    ( List.fold_left
+                                        reviews_info.approved_reviews
+                                        ~init:"Reviewed-by:\n" ~f:(fun s r ->
+                                          s ^ f "- %s\n" r)
+                                    ^
+                                    if
+                                      not
+                                        (List.is_empty
+                                           reviews_info.comment_reviews)
+                                    then
+                                      List.fold_left
+                                        reviews_info.comment_reviews
+                                        ~init:"Ack-by:\n" ~f:(fun s r ->
+                                          s ^ f "- %s\n" r)
+                                    else "" )
+                                  ~merge_method:`MERGE
+                                >>= fun () ->
+                                match
+                                  List.fold_left ~init:[] reviews_info.files
+                                    ~f:(fun acc f ->
+                                      if
+                                        string_match
+                                          ~regexp:
+                                            "dev/ci/user-overlays/\\(.*\\)" f
+                                      then Str.matched_group 1 f :: acc
+                                      else acc)
+                                with
+                                | [] ->
+                                    Lwt.return ()
+                                | overlays ->
+                                    GitHub_mutations.post_comment ~bot_info
+                                      ~message:
+                                        (f
+                                           "@%s: Please take care of the \
+                                            following overlays:\n\
+                                            %s"
+                                           comment_info.author
+                                           (List.fold_left overlays ~init:""
+                                              ~f:(fun s o -> s ^ f "- %s\n" o)))
+                                      ~id:pr.id )
+                            | Error _ ->
+                                GitHub_mutations.post_comment ~bot_info
+                                  ~message:
+                                    (f
+                                       "@%s: You can't merge this PR as you're \
+                                        not a member of the Pushers team."
+                                       comment_info.author)
+                                  ~id:pr.id ) ) )
+              | Error e ->
+                  GitHub_mutations.post_comment ~bot_info
+                    ~message:(f "@%s: %s" comment_info.author e)
```

Comment:
```suggestion
                    ~message:(f Something unexpected happend: %s\ncc @coq/coqbot-maintainers" comment_info.author e)
```

In your test, it looks like this comment was actually routinely posted until the PR got approved. We shouldn't rely on this line in case the PR is not approved, nor has a review request nor has a request for changes. Therefore, I suggest adding a fourth constructor to the `review_decision` type and treating `None` as an accepted case, not an error (encoded by this new constructor).